### ISSUE TITLE
ObservationLevel: Object?

### ISSUE THREAD

**Issue by ddachs** [tdwg/camtrap-dp#390]
I am curious if we are overlooking the (so far) most detailed observation level, specifically one that pertains to objects within media files. Currently, CamtrapDP supports two levels: media and event. When identifying objects in a media file (e.g., using MegaDetector), the count in the observation table will always be 1. For this reason, we moved from the media level to a more granular object level. I believe this distinction is crucial when generating count values for events, as having data at the media or object level call for different approaches.

**Comment by peterdesmet** [comment]
Can you clarify you question? Here's an attempt at providing info 😄 

1. It is possible to express object level observations in Camtrap DP. For that you use media-based observation (observationLevel = `media`, mediaID = not NULL) and use using [bboxX](https://camtrap-dp.tdwg.org/data/#observations.bboxX), [bboxY](https://camtrap-dp.tdwg.org/data/#observations.bboxY), [bboxWidth](https://camtrap-dp.tdwg.org/data/#observations.bboxWidth), [bboxHeight](https://camtrap-dp.tdwg.org/data/#observations.bboxHeight) to indicate where the object was observed. All remaining properties in the observation then apply to that object. Example:

https://github.com/tdwg/camtrap-dp/blob/6a903e9b34e05cbd54f04541161e849a1c3d2108/example/observations.csv?plain=1#L509-L512

When can be used to draw the bounding box: https://camtrap-dp.tdwg.org/example/62c200a9/#7245a2aa

2. Regarding how `individualCount` should be summed (without over-summing), we currently state the following in [observationLevel](https://camtrap-dp.tdwg.org/data/#observations.observationLevel):

> Level at which the observation was classified. `media` for media-based observations that are directly associated with a media file (`mediaID`). These are especially useful for machine learning and don't need to be mutually exclusive (e.g. multiple classifications are allowed). `event` for event-based observations that consider an event (comprising a collection of media files). These are especially useful for ecological research and should be mutually exclusive, so that their `count` can be summed.

On point 2, we may be moving towards an approach where `observations.csv` contains the biologically relevant information (one truth, with clear approach on how to create events). While for machine learning, it is useful to have _all_ classifications (multiple truths). See a very early draft proposal at https://github.com/tdwg/camtrap-dp/pull/389

**Comment by ddachs** [comment]
@kbubnicki correct!

**Comment by peterdesmet** [comment]
Makes sense to expand it to `object`, but we should think about terminology:

- Spatial region of interest: `object`?
- Temporal region of interest (indicated with `eventStart/eventEnd`), useful in video (and maybe later audio) files: `fragment`?
- Is it necessary to differentiate between the two in `observationLevel`? Is `submedia` enough?
- Are combinations of the spatial and temporal possible? Theoretically yes, if you use an temporal interval that identifies a frame in a video and then indicate the object. Or is it expected that for such use cases, the video is converted to images?

**Comment by ddachs** [comment]
1. **object** is a spatial region in a picture or video frame.
2. a temporal region to me in clearly an **interval**.  (I got used to [lubridates terminology](https://lubridate.tidyverse.org/reference/lubridate-package.html#timespans)) 
3. good question regarding submedia: On the one hand submedia keeps the hierarchichal terminology (event > media > submedia), but on the other hand, lacks the information what it exactly is. I prefer the differentiation. 
4. I hope, that the combination is possible. Our observation tables will explode, if we split all videos into single frames.

**Comment by ddachs** [comment]
In this context, I would prioritize the object rather than the interval, as the object is of primary interest. The combination of objects and their temporal presence is particularly relevant in videos, where object tracking is a key technology. Object tracking involves linking the spatial information of objects across individual video frames over time (temporal dimension). Reflecting on this, it becomes clear that we will need to store comprehensive information about **where** and **when** an object appears in the video. The exact format will likely depend on the conventions used in Computer Vision for handling such data. Perhaps a vector of bounding box information, with its length corresponding to the number of frames in the video, could be a suitable approach?

**Comment by peterdesmet** [comment]
@ddachs would a eventStart/eventEnd be sufficient to locate **when** an object appears in a video?
